exercises/runLengthDecode/runLengthDecode.js

```diff
@@ -10,11 +10,20 @@
  * @returns {string} The original, uncompressed string
  */
 
-function runLengthDecode(num) {
-  // This is your job. :)
-
-  // Remember, if the code is stumping you, take a step back and
-  // make sure you can do it by hand.
+function runLengthDecode(string) {
```

Comment:
This is actually a perfect use case for regular expressions.  The regex `/(\d+)([^\d_]+)/g` will match any sequence of digits followed by any sequence of non-digits, capturing each sequence.

Here's something you can try to pick apart:

```javascript
function runLengthDecode(string) {
  return Array.from(string.matchAll(/(\d+)([^\d]+)/g)).reduce((res, [_, count, char]) => {
    return res + char.repeat(Number(count));
  }, '');
}
```

Reply:
See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment for what is going on with

```javascript
(res, [_, count, char]) => { ... }
```